SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStrategySystem.cpp

```diff
@@ -0,0 +1,125 @@
+
+namespace SpatialGDK
+{
+void SpatialStrategySystem::Advance(SpatialOSWorkerInterface* Connection)
+{
+	// Messages are inspected to take care of the Worker entity creation.
+	const TArray<Worker_Op>& Messages = Connection->GetWorkerMessages();
+	for (const auto& Message : Messages)
+	{
+		switch (Message.op_type)
+		{
+		case WORKER_OP_TYPE_RESERVE_ENTITY_IDS_RESPONSE:
+		{
+			const Worker_ReserveEntityIdsResponseOp& Op = Message.op.reserve_entity_ids_response;
+			if (Op.request_id == StrategyWorkerRequest)
+			{
+				if (Op.first_entity_id == SpatialConstants::INVALID_ENTITY_ID)
+				{
+					UE_LOG(LogSpatialStrategySystem, Error, TEXT("Reserve entity failed : %s"), UTF8_TO_TCHAR(Op.message));
+					StrategyWorkerRequest = 0;
+				}
+				else
+				{
+					StrategyPartition = Message.op.reserve_entity_ids_response.first_entity_id;
+					CreateStrategyPartition(Connection);
+				}
+			}
+			break;
+		}
+		case WORKER_OP_TYPE_CREATE_ENTITY_RESPONSE:
+		{
+			const Worker_CreateEntityResponseOp& Op = Message.op.create_entity_response;
+			if (Op.request_id == StrategyWorkerRequest)
+			{
+				if (Op.entity_id == SpatialConstants::INVALID_ENTITY_ID)
+				{
+					UE_LOG(LogSpatialStrategySystem, Error, TEXT("Create entity failed : %s"), UTF8_TO_TCHAR(Op.message));
+				}
+				StrategyWorkerRequest = 0;
+
+				Worker_CommandRequest ClaimRequest = Worker::CreateClaimPartitionRequest(StrategyPartition);
+				StrategyWorkerRequest =
+					Connection->SendCommandRequest(StrategyWorkerSystemEntityId, &ClaimRequest, SpatialGDK::RETRY_UNTIL_COMPLETE, {});
+			}
+			break;
+		}
+		case WORKER_OP_TYPE_COMMAND_RESPONSE:
+		{
+			const Worker_CommandResponseOp& Op = Message.op.command_response;
+			if (Op.request_id == StrategyWorkerRequest)
+			{
+				if (Op.status_code != WORKER_STATUS_CODE_SUCCESS)
+				{
+					UE_LOG(LogSpatialStrategySystem, Error, TEXT("Claim partition failed : %s"), UTF8_TO_TCHAR(Op.message));
+				}
+				StrategyWorkerRequest = 0;
+			}
+		}
+		break;
+		}
+	}
+
+	const FSubViewDelta& SubViewDelta = SubView.GetViewDelta();
+	for (const EntityDelta& Delta : SubViewDelta.EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::UPDATE:
+		{
+			// TODO
+		}
+		break;
+		case EntityDelta::ADD:
+		{
+			// TODO
+		}
+		break;
+		case EntityDelta::REMOVE:
+		case EntityDelta::TEMPORARILY_REMOVED:
+		{
+			// TODO
+		}
+		break;
+		default:
+			break;
+		}
+	}
+}
+
+void SpatialStrategySystem::Flush(SpatialOSWorkerInterface* Connection)
+{
+	// TODO
+}
+
+void SpatialStrategySystem::Init(SpatialOSWorkerInterface* Connection)
+{
+	StrategyWorkerRequest = Connection->SendReserveEntityIdsRequest(1, SpatialGDK::RETRY_UNTIL_COMPLETE);
+}
+
+void SpatialStrategySystem::CreateStrategyPartition(SpatialOSWorkerInterface* Connection)
+{
+	AuthorityDelegationMap Map;
+	Map.Add(SpatialConstants::GDK_KNOWN_ENTITY_AUTH_COMPONENT_SET_ID, StrategyPartition);
+
+	TArray<FWorkerComponentData> Components;
```

Comment:
We might not need that going forward, I expect the routing and strategy partition to be statically assigned when creating the initial snapshot.
That was something I kind of hacked because the partitions arrived in the middle of my dev on the routing worker, but we should not retain that approach.

Reply:
One of the acceptance criteria of the task is: "creates a server-entity", I'm not sure on the terminology but isn't that what we are doing here?

Reply:
"create a server entity" is only a little bit of the complete story, this acceptance criteria does not tell us much.
Now, we actually do need some entities for 1) authority partition 2) interest
The part I'm not happy about is that other workers need to know that partition, and creating it dynamically forces us to go through the query path, which could be replaced by a statically created entity in the initial snapshot. The interest can be created dynamically, or in the initial snapshot, this one does not make much difference.

Reply:
I've changed Strategy Partition Entity to be statically created in the snapshot rather than dynamically.